src/rust/bitbox02/src/lib.rs

```diff
@@ -156,6 +155,21 @@ pub fn reset(status: bool) {
     unsafe { bitbox02_sys::reset_reset(status) }
 }
 
+pub fn strftime(timestamp: u32, format: &str) -> String {
+    let mut out = [0u8; 100];
+    unsafe {
+        bitbox02_sys::strftime(
+            out.as_mut_ptr(),
+            out.len() as _,
+            crate::util::str_to_cstr_vec(format).unwrap().as_ptr(),
```

Comment:
You had actually taught me that this is safe :smile: 

https://github.com/digitalbitbox/bitbox02-firmware/pull/603#discussion_r499138259

Not sure about propagation - I don't mind panics so much if the error is supposed to never happen (programming errors). If it happens in the wild, the error will be even shown on the screen, otherwise the API call will just return a generic error (a point that could be improved, but it's not easy)

Reply:
I got a bit scared because you were taking a raw pointer to the string. But I can see now that the pointer is never used outside of this function call. So I'm also pretty convinced that this is OK.